source/org/zfin/mutant/repository/HibernateMutantRepository.java

```diff
@@ -276,97 +271,102 @@ public PaginationResult<FishGenotypeFeature> getFishByFeature(String featureId,
+                      when zyg_name = 'unknown' then 3 
+                      when zyg_name = 'complex' then 4 
+                      else 5 
+                    end 
+                  ) asc, 
+                  affected_count asc, 
+                  affected_list asc, 
+                  fish_name_order asc;
+                """;
+
+        Query query = currentSession().createSQLQuery(sql);
         query.setParameter("feature_id", featureId);
-        query.setResultTransformer(new BasicTransformerAdapter() {
-            @Override
-            public FishGenotypeFeature transformTuple(Object[] objects, String[] strings) {
-                String genotypeFeatureId = (String) objects[0];
-                String fishId = (String) objects[1];
 
-                GenotypeFeature genotypeFeature = getGenotypeFeature(genotypeFeatureId);
-                Fish fish = getFish(fishId);
+        setTupleResultTransformer(query, (Object[] tuples, String[] aliases) -> {
+            String genotypeFeatureId = (String) tuples[0];
+            String fishId = (String) tuples[1];
 
-                return new FishGenotypeFeature(fish, genotypeFeature);
-            }
+            GenotypeFeature genotypeFeature = getGenotypeFeature(genotypeFeatureId);
+            Fish fish = getFish(fishId);
+
+            return new FishGenotypeFeature(fish, genotypeFeature);
         });
 
         return PaginationResultFactory.createResultFromScrollableResultAndClose(pagination.getStart(), pagination.getEnd(), query.scroll());
     }
 
     public int getNumberOfImagesPerAnatomyAndMutant(GenericTerm term, Genotype genotype) {
-        Session session = HibernateUtil.currentSession();
-
-        String hql = "select count(distinct image) from Image image, Figure fig, ExpressionResult res, " +
-            "                                  ExpressionExperiment exp " +
-            "where " +
-            "res member of exp.expressionResults AND " +
-            "res.entity.superterm = :term AND " +
-            "fig member of res.figures AND " +
-            "image member of fig.images AND " +
-            "res.expressionFound = :expressionFound AND " +
-            "exp.fishExperiment.fish.genotype.zdbID = :genoZdbID ";
-        Query query = session.createQuery(hql);
-        query.setBoolean("expressionFound", true);
+        Session session = currentSession();
+
+        String hql = """
+             select count(distinct image) from Image image, Figure fig, ExpressionResult res, 
+                                              ExpressionExperiment exp 
+            where 
+            res member of exp.expressionResults AND 
+            res.entity.superterm = :term AND 
+            fig member of res.figures AND 
+            image member of fig.images AND 
+            res.expressionFound = true AND 
+            exp.fishExperiment.fish.genotype.zdbID = :genoZdbID 
+            """;
+        Query query = session.createQuery(hql, Number.class);
```

Comment:
use the generic Query class, i.e.
Query<Number> query =session.createQuery(hql, Number.class);

Reply:
Sorry, I'm not seeing the distinction?